osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySliderBall.cs

```diff
@@ -1,64 +1,77 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using osu.Framework.Allocation;
 using osu.Framework.Bindables;
+using osu.Framework.Extensions.ObjectExtensions;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Sprites;
 using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Osu.Objects.Drawables;
 using osu.Game.Skinning;
+using osuTK;
 using osuTK.Graphics;
 
 namespace osu.Game.Rulesets.Osu.Skinning.Legacy
 {
     public partial class LegacySliderBall : CompositeDrawable
     {
-        private readonly Drawable animationContent;
-
         private readonly ISkin skin;
 
         [Resolved(canBeNull: true)]
         private DrawableHitObject? parentObject { get; set; }
 
-        public Color4 BallColour => animationContent.Colour;
-
         private Sprite layerNd = null!;
+        private Drawable ball = null!;
         private Sprite layerSpec = null!;
 
-        public LegacySliderBall(Drawable animationContent, ISkin skin)
+        public Color4 BallColour => ball.Colour;
+
+        public LegacySliderBall(ISkin skin)
         {
-            this.animationContent = animationContent;
             this.skin = skin;
 
             AutoSizeAxes = Axes.Both;
         }
 
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(DrawableHitObject drawableHitObject)
```

Comment:
there is already a `[Resolved] DrawableHitObject` field in this class, and it's nullable too, while this isn't...?

Reply:
I...think there should be checks against a dependency being resolved both in BDL and as a `Resolved` property.